File: SingletonDesignPattern/Logger_Example/SOL1_lazySingleton/Driver.java
package SingletonDesignPattern.Logger_Example.SOL1_lazySingleton;

public class Driver {

  public static void main(String[] args) {
    Logger logger1 = Logger.getInstance();
    Logger logger2 = Logger.getInstance();

    logger1.Log("This is the Log from logger instance 1.");
    logger2.Log("This is the Log from logger instance 2.");
  }
}

File: SingletonDesignPattern/Logger_Example_Python/SOL2_DoubleCheckLocking_MetaClass/Driver.py
# import threading
from Logger import Logger

class Driver():

  @staticmethod
  def main():

    logger1 = Logger()
    logger2 = Logger()

    logger1.Log("This is the log from the object logger1")
    logger2.Log("This is the log from the object logger2")

    print(logger1 == logger2)

    # thread1 = threading.Thread(target=lambda: Logger().Log("This is the log from the LoggerObject created by Thread1 "))
    # thread2 = threading.Thread(target=lambda: Logger().Log("This is the log from the LoggerObject created by Thread2 "))
    # thread1.start()
    # thread2.start()
    # thread1.join()
    # thread2.join()

if __name__ == "__main__":
  Driver.main()